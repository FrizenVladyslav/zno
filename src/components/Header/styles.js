import { StyleSheet } from 'aphrodite/no-important'

export default StyleSheet.create({
  menu: {
    padding: '0 20px',
    marginTop: 0,
    marginBottom: 30,
  },
})
