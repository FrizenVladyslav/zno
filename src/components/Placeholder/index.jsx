import React from 'react'
import PropTypes from 'prop-types'
import { Placeholder } from 'semantic-ui-react'

const PlaceholderGenerator = ({ paragraphs }) => {
  return (
    <Placeholder>
      {paragraphs.map((line, index) => (
        <Placeholder.Paragraph key={index}>
          {[...Array(line)].map((l, index) => (
            <Placeholder.Line key={index} />
          ))}
        </Placeholder.Paragraph>
      ))}
    </Placeholder>
  )
}

PlaceholderGenerator.propTypes = {
  paragraphs: PropTypes.arrayOf(PropTypes.number),
}

export default PlaceholderGenerator
