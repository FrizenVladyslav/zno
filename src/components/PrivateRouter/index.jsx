import React from 'react'
import { Route, Redirect } from 'react-router'
import { toast } from 'react-toastify'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const PrivateRoute = props => {
  let allow = false

  // here private logic ...
  allow = props.roles.includes(props.role)

  // show a notification
  if (!allow) toast.info(props.message || `You have not access to this page`)

  return allow ? (
    <Route {...props} />
  ) : (
    <Redirect to={props.redirectTo ? props.redirectTo : '/'} />
  )
}

PrivateRoute.propTypes = {
  roles: PropTypes.arrayOf(PropTypes.string).isRequired,
  message: PropTypes.string,
  redirectTo: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
}

// @ts-ignore
export default connect(({ user }) => ({ role: user.role }))(PrivateRoute)
