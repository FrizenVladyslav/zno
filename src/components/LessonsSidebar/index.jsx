import React from 'react'
import { Menu, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite'

import styles from './styles'

const Sidebar = () => {
  return (
    <Menu vertical className={css(styles.sidebar)}>
      <Menu.Item header>Предмет</Menu.Item>
      <Menu.Item name="Українська мова" />
    </Menu>
  )
}

export default Sidebar
