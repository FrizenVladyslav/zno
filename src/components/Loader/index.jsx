import React from 'react'
import PropTypes from 'prop-types'
import { Loader as SemanticLoader } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'

import styles from './styles'

const Loader = props => {
  return (
    <div
      className={css(styles.loaderContainer)}
      style={{ minHeight: props.height }}
    >
      <SemanticLoader active inline="centered" />
    </div>
  )
}

Loader.propTypes = {
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

export default Loader
