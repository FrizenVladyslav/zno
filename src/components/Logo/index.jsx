import React from 'react'

import { Link } from 'react-router-dom'
import { Icon } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import styles from './styles'

const Logo = props => {
  return (
    <Link to="/" className={css(styles.logo)}>
      <Icon name="student" size="huge" />
      <h2>Краще за всіх</h2>
    </Link>
  )
}

export default Logo
