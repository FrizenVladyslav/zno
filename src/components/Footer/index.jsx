import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Container, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import Logo from '../Logo'

import styles from './styles'

const Footer = props => {
  return (
    <Segment placeholder className={css(styles.footer)}>
      <Container>
        <div>
          <Logo style={{ alignSelf: 'flex-end' }} />
          <p>&copy; Всі права захищені)</p>
        </div>
        <ul>
          <li>Предмети:</li>
          {props.lessons &&
            props.lessons.map(lesson => (
              <li key={lesson._id}>
                <Link to={`/lessons/${lesson._id}`}>{lesson.title}</Link>
              </li>
            ))}
        </ul>
        <ul>
          <li>Інші посилання:</li>
          <li>
            <Link to="/login">Увійти</Link>
          </li>
          <li>
            <Link to="/lessons">Зайняття</Link>
          </li>
        </ul>
      </Container>
    </Segment>
  )
}

// @ts-ignore
export default connect(({ lesson }) => ({ lessons: lesson.lessons }))(Footer)
