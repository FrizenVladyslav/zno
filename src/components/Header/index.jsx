import React, { useState } from 'react'
import { connect } from 'react-redux'
import { Menu } from 'semantic-ui-react'
import { NavLink } from 'react-router-dom'
import { css } from 'aphrodite'
import Logo from '../Logo'
import ConfirmModal from '../Modals/ConfirmModal'

import * as userActions from 'actions/user'

import styles from './styles'

const Header = props => {
  const [modal, triggerModal] = useState(false)

  return (
    <>
      <Menu stackable className={css(styles.menu)}>
        <Menu.Header>
          <Logo />
        </Menu.Header>
        <Menu.Menu position="right">
          {props.user.role === 'admin' && (
            <Menu.Item name="admin" link as={NavLink} to="/admin">
              Адміністрування
            </Menu.Item>
          )}
          <Menu.Item name="lessons" link as={NavLink} to="/lessons">
            Предмети
          </Menu.Item>
          {!!props.user._id ? (
            <Menu.Item link onClick={() => triggerModal(true)}>
              {props.user.nick}
            </Menu.Item>
          ) : (
            <Menu.Item as={NavLink} link name="sign-in" to="/login">
              Увійти
            </Menu.Item>
          )}
        </Menu.Menu>
      </Menu>
      <ConfirmModal
        icon="sign-out"
        message="Ви дійсно бажаєте вийти?"
        open={modal}
        onClose={() => triggerModal(false)}
        onSuccess={userActions.signOut}
        title="Вийти"
      />
    </>
  )
}

// @ts-ignore
export default connect(({ user }) => ({ user }))(Header)
