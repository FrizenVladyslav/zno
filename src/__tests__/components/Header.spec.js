import React from 'react'
import { shallow } from 'enzyme'
import { Header } from 'components/Header'
import { StyleSheetTestUtils } from 'aphrodite'

describe('header', () => {
  afterEach(() => {
    return new Promise(resolve => {
      StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
      return process.nextTick(resolve)
    })
  })

  const initUser = {
    _id: 'id',
    nick: 'nick',
  }

  describe('if user unlogin', () => {
    const header = shallow(<Header user={{}} />)
    it('renders properly', () => {
      expect(header).toMatchSnapshot()
    })

    it('menu have 2 items', () => {
      expect(header.find('MenuItem')).toHaveLength(2)
    })

    it('menu have login', () => {
      expect(
        header
          .find('MenuItem')
          .last()
          .props().children
      ).toBe('Увійти')
    })
  })

  describe('if user login', () => {
    const header = shallow(<Header user={initUser} />)
    it('renders properly', () => {
      expect(header).toMatchSnapshot()
    })

    it('menu have 2 items', () => {
      expect(header.find('MenuItem')).toHaveLength(2)
    })

    it('menu have logout', () => {
      expect(
        header
          .find('MenuItem')
          .last()
          .props().children
      ).toBe(initUser.nick)
    })
  })

  describe('if user admin', () => {
    const header = shallow(<Header user={{ ...initUser, role: 'admin' }} />)
    it('renders properly', () => {
      expect(header).toMatchSnapshot()
    })

    it('menu have 3 items', () => {
      expect(header.find('MenuItem')).toHaveLength(3)
    })
  })
})
