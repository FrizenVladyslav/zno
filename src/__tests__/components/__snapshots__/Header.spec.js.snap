// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header if user admin renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={
      Object {
        "_id": "id",
        "nick": "nick",
        "role": "admin",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Menu
          className="menu_101l2md"
          stackable={true}
        >
          <MenuHeader>
            <Logo />
          </MenuHeader>
          <MenuMenu
            position="right"
          >
            <MenuItem
              as={[Function]}
              link={true}
              name="admin"
              to="/admin"
            >
              Адміністрування
            </MenuItem>
            <MenuItem
              as={[Function]}
              link={true}
              name="lessons"
              to="/lessons"
            >
              Предмети
            </MenuItem>
            <MenuItem
              link={true}
              onClick={[Function]}
            >
              nick
            </MenuItem>
          </MenuMenu>
        </Menu>,
        <ConfirmModal
          icon="sign-out"
          message="Ви дійсно бажаєте вийти?"
          onClose={[Function]}
          onSuccess={[Function]}
          open={false}
          title="Вийти"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuHeader>
              <Logo />
            </MenuHeader>,
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="admin"
                to="/admin"
              >
                Адміністрування
              </MenuItem>
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                link={true}
                onClick={[Function]}
              >
                nick
              </MenuItem>
            </MenuMenu>,
          ],
          "className": "menu_101l2md",
          "stackable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Logo />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="admin"
                  to="/admin"
                >
                  Адміністрування
                </MenuItem>,
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>,
                <MenuItem
                  link={true}
                  onClick={[Function]}
                >
                  nick
                </MenuItem>,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": [Function],
                  "children": "Адміністрування",
                  "link": true,
                  "name": "admin",
                  "to": "/admin",
                },
                "ref": null,
                "rendered": "Адміністрування",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": [Function],
                  "children": "Предмети",
                  "link": true,
                  "name": "lessons",
                  "to": "/lessons",
                },
                "ref": null,
                "rendered": "Предмети",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "nick",
                  "link": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "nick",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "sign-out",
          "message": "Ви дійсно бажаєте вийти?",
          "onClose": [Function],
          "onSuccess": [Function],
          "open": false,
          "title": "Вийти",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Menu
            className="menu_101l2md"
            stackable={true}
          >
            <MenuHeader>
              <Logo />
            </MenuHeader>
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="admin"
                to="/admin"
              >
                Адміністрування
              </MenuItem>
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                link={true}
                onClick={[Function]}
              >
                nick
              </MenuItem>
            </MenuMenu>
          </Menu>,
          <ConfirmModal
            icon="sign-out"
            message="Ви дійсно бажаєте вийти?"
            onClose={[Function]}
            onSuccess={[Function]}
            open={false}
            title="Вийти"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuHeader>
                <Logo />
              </MenuHeader>,
              <MenuMenu
                position="right"
              >
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="admin"
                  to="/admin"
                >
                  Адміністрування
                </MenuItem>
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>
                <MenuItem
                  link={true}
                  onClick={[Function]}
                >
                  nick
                </MenuItem>
              </MenuMenu>,
            ],
            "className": "menu_101l2md",
            "stackable": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Logo />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <MenuItem
                    as={[Function]}
                    link={true}
                    name="admin"
                    to="/admin"
                  >
                    Адміністрування
                  </MenuItem>,
                  <MenuItem
                    as={[Function]}
                    link={true}
                    name="lessons"
                    to="/lessons"
                  >
                    Предмети
                  </MenuItem>,
                  <MenuItem
                    link={true}
                    onClick={[Function]}
                  >
                    nick
                  </MenuItem>,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": "Адміністрування",
                    "link": true,
                    "name": "admin",
                    "to": "/admin",
                  },
                  "ref": null,
                  "rendered": "Адміністрування",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": "Предмети",
                    "link": true,
                    "name": "lessons",
                    "to": "/lessons",
                  },
                  "ref": null,
                  "rendered": "Предмети",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "nick",
                    "link": true,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "nick",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "sign-out",
            "message": "Ви дійсно бажаєте вийти?",
            "onClose": [Function],
            "onSuccess": [Function],
            "open": false,
            "title": "Вийти",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`header if user login renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={
      Object {
        "_id": "id",
        "nick": "nick",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Menu
          className="menu_101l2md"
          stackable={true}
        >
          <MenuHeader>
            <Logo />
          </MenuHeader>
          <MenuMenu
            position="right"
          >
            <MenuItem
              as={[Function]}
              link={true}
              name="lessons"
              to="/lessons"
            >
              Предмети
            </MenuItem>
            <MenuItem
              link={true}
              onClick={[Function]}
            >
              nick
            </MenuItem>
          </MenuMenu>
        </Menu>,
        <ConfirmModal
          icon="sign-out"
          message="Ви дійсно бажаєте вийти?"
          onClose={[Function]}
          onSuccess={[Function]}
          open={false}
          title="Вийти"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuHeader>
              <Logo />
            </MenuHeader>,
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                link={true}
                onClick={[Function]}
              >
                nick
              </MenuItem>
            </MenuMenu>,
          ],
          "className": "menu_101l2md",
          "stackable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Logo />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                false,
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>,
                <MenuItem
                  link={true}
                  onClick={[Function]}
                >
                  nick
                </MenuItem>,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": [Function],
                  "children": "Предмети",
                  "link": true,
                  "name": "lessons",
                  "to": "/lessons",
                },
                "ref": null,
                "rendered": "Предмети",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "nick",
                  "link": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "nick",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "sign-out",
          "message": "Ви дійсно бажаєте вийти?",
          "onClose": [Function],
          "onSuccess": [Function],
          "open": false,
          "title": "Вийти",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Menu
            className="menu_101l2md"
            stackable={true}
          >
            <MenuHeader>
              <Logo />
            </MenuHeader>
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                link={true}
                onClick={[Function]}
              >
                nick
              </MenuItem>
            </MenuMenu>
          </Menu>,
          <ConfirmModal
            icon="sign-out"
            message="Ви дійсно бажаєте вийти?"
            onClose={[Function]}
            onSuccess={[Function]}
            open={false}
            title="Вийти"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuHeader>
                <Logo />
              </MenuHeader>,
              <MenuMenu
                position="right"
              >
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>
                <MenuItem
                  link={true}
                  onClick={[Function]}
                >
                  nick
                </MenuItem>
              </MenuMenu>,
            ],
            "className": "menu_101l2md",
            "stackable": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Logo />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  false,
                  <MenuItem
                    as={[Function]}
                    link={true}
                    name="lessons"
                    to="/lessons"
                  >
                    Предмети
                  </MenuItem>,
                  <MenuItem
                    link={true}
                    onClick={[Function]}
                  >
                    nick
                  </MenuItem>,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": "Предмети",
                    "link": true,
                    "name": "lessons",
                    "to": "/lessons",
                  },
                  "ref": null,
                  "rendered": "Предмети",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "nick",
                    "link": true,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "nick",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "sign-out",
            "message": "Ви дійсно бажаєте вийти?",
            "onClose": [Function],
            "onSuccess": [Function],
            "open": false,
            "title": "Вийти",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`header if user unlogin renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Menu
          className="menu_101l2md"
          stackable={true}
        >
          <MenuHeader>
            <Logo />
          </MenuHeader>
          <MenuMenu
            position="right"
          >
            <MenuItem
              as={[Function]}
              link={true}
              name="lessons"
              to="/lessons"
            >
              Предмети
            </MenuItem>
            <MenuItem
              as={[Function]}
              link={true}
              name="sign-in"
              to="/login"
            >
              Увійти
            </MenuItem>
          </MenuMenu>
        </Menu>,
        <ConfirmModal
          icon="sign-out"
          message="Ви дійсно бажаєте вийти?"
          onClose={[Function]}
          onSuccess={[Function]}
          open={false}
          title="Вийти"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuHeader>
              <Logo />
            </MenuHeader>,
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                as={[Function]}
                link={true}
                name="sign-in"
                to="/login"
              >
                Увійти
              </MenuItem>
            </MenuMenu>,
          ],
          "className": "menu_101l2md",
          "stackable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Logo />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                false,
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>,
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="sign-in"
                  to="/login"
                >
                  Увійти
                </MenuItem>,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": [Function],
                  "children": "Предмети",
                  "link": true,
                  "name": "lessons",
                  "to": "/lessons",
                },
                "ref": null,
                "rendered": "Предмети",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": [Function],
                  "children": "Увійти",
                  "link": true,
                  "name": "sign-in",
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Увійти",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "sign-out",
          "message": "Ви дійсно бажаєте вийти?",
          "onClose": [Function],
          "onSuccess": [Function],
          "open": false,
          "title": "Вийти",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Menu
            className="menu_101l2md"
            stackable={true}
          >
            <MenuHeader>
              <Logo />
            </MenuHeader>
            <MenuMenu
              position="right"
            >
              <MenuItem
                as={[Function]}
                link={true}
                name="lessons"
                to="/lessons"
              >
                Предмети
              </MenuItem>
              <MenuItem
                as={[Function]}
                link={true}
                name="sign-in"
                to="/login"
              >
                Увійти
              </MenuItem>
            </MenuMenu>
          </Menu>,
          <ConfirmModal
            icon="sign-out"
            message="Ви дійсно бажаєте вийти?"
            onClose={[Function]}
            onSuccess={[Function]}
            open={false}
            title="Вийти"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuHeader>
                <Logo />
              </MenuHeader>,
              <MenuMenu
                position="right"
              >
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="lessons"
                  to="/lessons"
                >
                  Предмети
                </MenuItem>
                <MenuItem
                  as={[Function]}
                  link={true}
                  name="sign-in"
                  to="/login"
                >
                  Увійти
                </MenuItem>
              </MenuMenu>,
            ],
            "className": "menu_101l2md",
            "stackable": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Logo />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  false,
                  <MenuItem
                    as={[Function]}
                    link={true}
                    name="lessons"
                    to="/lessons"
                  >
                    Предмети
                  </MenuItem>,
                  <MenuItem
                    as={[Function]}
                    link={true}
                    name="sign-in"
                    to="/login"
                  >
                    Увійти
                  </MenuItem>,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": "Предмети",
                    "link": true,
                    "name": "lessons",
                    "to": "/lessons",
                  },
                  "ref": null,
                  "rendered": "Предмети",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": "Увійти",
                    "link": true,
                    "name": "sign-in",
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Увійти",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "sign-out",
            "message": "Ви дійсно бажаєте вийти?",
            "onClose": [Function],
            "onSuccess": [Function],
            "open": false,
            "title": "Вийти",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
